src/parser/statement.js

```diff
@@ -913,6 +937,20 @@ pp.checkExport = function (node) {
   }
 };
 
+pp.checkDuplicateExports = function(node, name, isDefault) {
+  if (this.state.exportedIdentifiers[name]) {
```

Comment:
Oh wow, that makes sense. @danez Let me know if you need me work on a fix - happy to do it.